framework/db/ActiveRecord.php

```diff
@@ -454,20 +454,15 @@ protected function insertInternal($attributes = null)
                 $values[$key] = $value;
             }
         }
-        $db = static::getDb();
-        $command = $db->createCommand()->insert($this->tableName(), $values);
-        if (!$command->execute()) {
+        if (($primaryKeys = static::getDb()->schema->insert($this->tableName(), $values)) === false) {
             return false;
         }
-        $table = $this->getTableSchema();
-        if ($table->sequenceName !== null) {
-            foreach ($table->primaryKey as $name) {
-                if ($this->getAttribute($name) === null) {
-                    $id = $table->columns[$name]->phpTypecast($db->getLastInsertID($table->sequenceName));
-                    $this->setAttribute($name, $id);
-                    $values[$name] = $id;
-                    break;
-                }
+        foreach ($primaryKeys as $name => $value) {
+            if ($this->getAttribute($name) === null) {
```

Comment:
I've improved that check when moving code from AR to `Schema::insert` in my PR. Previously it assigned last insert id to first null PK column. I've changed it to assign it to first auto incremented PK column.